### PR TITLE
[NVIDIA TF-TRT] Improvements of the TRTConverterV2 summary - Golden APIs Updates

### PR DESCRIPTION
**Pull request by DEKHTIARJonathan**
Minor improvements of the `TRTConverterV2.summary()` method.
- Deprecating `detailed=True` in favor of `verbosity_lvl=[0..2]`
- Prints more details according to the verbosity level requested.
- Default behavior unchanged

@poulsbo for final approval
@pavanimajety for review

Overwrite and replaces: #58076

### CODE REVIEW

**Comment by poulsbo** [comment]
The goal is to have @pavanimajety review this.

**Review by pjannaty — APPROVED** [review]
LGTM. Thanks, Jonathan! Do we know why the `Py+CPP Test Suite - Ubuntu CPU, Python 3.9 Expected — Waiting for status to be reported`?

**Comment by DEKHTIARJonathan** [comment]
@poulsbo it's been 2 weeks we are getting stalled. Have we made any progress ?
@pavanimajety still is not appointed as reviewer and we can't proceed forward

**Comment by pjannaty** [comment]
> @poulsbo it's been 2 weeks we are getting stalled. Have we made any progress ?
> @pavanimajety still is not appointed as reviewer and we can't proceed forward

Chatted with Jonathan offline and filled him in wrt the new setup.

File: tensorflow/python/compiler/tensorrt/trt_convert.py [hunk 1]
```diff
@@ -1606,7 +1606,13 @@ def _reset_allow_build_at_runtime(node):
...
     save.save(trackable, output_saved_model_dir, signatures, options=options)
 
-  def summary(self, line_length=160, detailed=True, print_fn=None):
+  @deprecation.deprecated_arg_values(
+      None,
+      "Consider using instead: `summary(..., verbosity_lvl=[0..2], ...)`",
+      warn_once=True,
+      detailed=True)
+  def summary(self, line_length=160, detailed=False, verbosity_lvl=1,
+              print_fn=None):
```
Comment (gbaned) [inline tensorflow/python/compiler/tensorrt/trt_convert.py:1615]: The new argument should be at the end to avoid breaking users who are not using keywords to specify the arguments. 

If someone used this API as follows: 'foo.summary(100, True, lambda x..)' before this change, then this change would break them.

File: tensorflow/tools/api/golden/v2/tensorflow.experimental.tensorrt.-converter.pbtxt [hunk 2]
```diff
@@ -4,7 +4,7 @@ tf_class {
   is_instance: "<type \'object\'>"
   member_method {
     name: "__init__"
-    argspec: "args=[\'self\', \'input_saved_model_dir\', \'input_saved_model_tags\', \'input_saved_model_signature_key\', \'use_dynamic_shape\', \'dynamic_shape_profile_strategy\', \'max_workspace_size_bytes\', \'precision_mode\', \'minimum_segment_size\', \'maximum_cached_engines\', \'use_calibration\', \'allow_build_at_runtime\', \'conversion_params\'], varargs=None, keywords=None, defaults=[\'None\', \'None\', \'None\', \'None\', \'None\', \'1073741824\', \'FP32\', \'3\', \'1\', \'True\', \'True\', \'None\'], "
+    argspec: "args=[\'self\', \'input_saved_model_dir\', \'input_saved_model_tags\', \'input_saved_model_signature_key\', \'use_dynamic_shape\', \'dynamic_shape_profile_strategy\', \'max_workspace_size_bytes\', \'precision_mode\', \'minimum_segment_size\', \'maximum_cached_engines\', \'use_calibration\', \'allow_build_at_runtime\', \'conversion_params\'], varargs=None, keywords=None, defaults=[\'None\', \'None\', \'None\', \'None\', \'None\', \'9223372036854775295\', \'FP32\', \'3\', \'1\', \'True\', \'True\', \'None\'], "
```
Comment (gbaned) [inline tensorflow/tools/api/golden/v2/tensorflow.experimental.tensorrt.-converter.pbtxt:7]: The change of the default value seems unrelated to this change.

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @DEKHTIARJonathan Any update on this PR? Please. Thank you!
